kroxylicious-runtime/src/main/java/io/kroxylicious/proxy/model/VirtualCluster.java

```diff
@@ -229,6 +259,45 @@ private static NettyTrustProvider configureTrustProvider(Tls tlsConfiguration) {
         return new NettyTrustProvider(trustProvider);
     }
 
+    private static void configureCipherSuites(SslContextBuilder sslContextBuilder, Tls tlsConfiguration) {
+        Optional.ofNullable(tlsConfiguration.cipherSuites())
+                .map(ciphers -> sslContextBuilder.ciphers(
+                        tlsConfiguration.cipherSuites().allowed(),
+                        new DenyCipherSuiteFilter(tlsConfiguration.cipherSuites().denied())));
+    }
+
+    private static void configureEnabledProtocols(SslContextBuilder sslContextBuilder, Tls tlsConfiguration) {
+
+        try {
+            var protocols = Optional.ofNullable(tlsConfiguration.protocols());
+
+            if (protocols.isPresent()) {
+                var allowedProtocols = Optional.ofNullable(protocols.get().allowed())
+                        .orElse(Arrays.stream(SSLContext.getDefault().getSupportedSSLParameters().getProtocols())
```

Comment:
Using `SSLContext.getDefault().getSupportedSSLParameters().getProtocols()` to get the default protocols only works if the user is using the JDK for TLS.    If the user has chosen to use the OpenSSL, then it is going to give the wrong answer.  See `io.netty.handler.ssl.OpenSsl#defaultProtocols(boolean)`.

That said, I think what you have is good enough for this PR. 

It is a pity Netty doesn't have a builder method (@franz1981 WDYT?)
```
SslContextBuilder protocols(Iterable<String> protocols, ProtocolFilter protocolFilter)
```